### ISSUE TITLE
batchsize and subdivision

### ISSUE THREAD

**Issue by Xiaohui9607** [DeNA/PyTorch_YOLOv3#14]
the default config set the batchsize to 4 and subdivision to 16, in darknet each time it takes batchsize/subdivision into training procedure. What the difference between your config and the darknet config?

**Comment by hirotomusiker** [comment]
https://github.com/pjreddie/darknet/blob/master/cfg/yolov3.cfg#L6-L7
It's a bit confusing.  In our setting, 'batchsize' corresponds to batchsize/subdivision of darknet.
We train 4 images at a time and accumulate the loss 16 times. It's the same procedure as darknet.
Thank you!

**Comment by LightToYang** [comment]
Should the gradient accumulation be averaged?
```
# subdivision loop
optimizer.zero_grad()
for inner_iter_i in range(subdivision):
     try:
        imgs, targets, _, _ = next(dataiterator)  # load a batch
     except StopIteration:
        dataiterator = iter(dataloader)
        imgs, targets, _, _ = next(dataiterator)  # load a batch
     imgs = Variable(imgs.type(dtype))
     targets = Variable(targets.type(dtype), requires_grad=False)
     loss = model(imgs, targets)

     loss = loss /subdivision 

     loss.backward()
optimizer.step()
```

**Comment by hirotomusiker** [comment]
No, it should not be, because the learning rate is divided by batchsize and subdivision : 
https://github.com/DeNA/PyTorch_YOLOv3/blob/master/train.py#L133
We follow the original implementation's way. The loss is the accumulation of the effective batch (in our case, subdivision * batchsize).

thank you!

**Comment by hirotomusiker** [comment]
I suppose this issue is resolved.
